functionaltests/steps.go

```diff
@@ -314,8 +337,97 @@ type customStep struct {
 	Func stepFunc
 }
 
-var _ testStep = customStep{}
-
 func (c customStep) Step(t *testing.T, ctx context.Context, e *testStepEnv, previousRes testStepResult) testStepResult {
 	return c.Func(t, ctx, e, previousRes)
 }
+
+// expectedDataStreamsIngest represent the expected number of ingested document
+// after a single run of ingest.
+//
+// NOTE: The aggregation data streams have negative counts, because they are
+// expected to appear but the document counts should not be asserted.
+func expectedDataStreamsIngest(namespace string) esclient.DataStreamsDocCount {
+	return map[string]int{
+		fmt.Sprintf("traces-apm-%s", namespace):                     15013,
+		fmt.Sprintf("metrics-apm.app.opbeans_python-%s", namespace): 1437,
+		fmt.Sprintf("metrics-apm.internal-%s", namespace):           1351,
+		fmt.Sprintf("logs-apm.error-%s", namespace):                 364,
+		// Ignore aggregation data streams.
+		fmt.Sprintf("metrics-apm.service_destination.1m-%s", namespace): -1,
+		fmt.Sprintf("metrics-apm.service_transaction.1m-%s", namespace): -1,
+		fmt.Sprintf("metrics-apm.service_summary.1m-%s", namespace):     -1,
+		fmt.Sprintf("metrics-apm.transaction.1m-%s", namespace):         -1,
+	}
```

Comment:
Would it not be enough to test for number of docs in datastreams > 0 or higher than at the start of a version test?